### DISCUSSION TITLE
Ubuntu WSL - Connection was reset

### DISCUSSION THREAD

**Discussion by axelthat** [vikejs/vike#1657] [Discussion]
### Description

I created a react project using vike. I haven't added anything. It's just a barebones project. I ran `npm run dev` inside WSL. And then when I goto the browser, it says "connection was reset".

![image](https://github.com/vikejs/vike/assets/31353695/c29bfdf2-6db3-400e-87d8-9e526fcb780d)

Ports are also forwarded correctly.

![image](https://github.com/vikejs/vike/assets/31353695/9a65a546-df13-46fc-b21b-9a765eb97e7c)

I also changed the host to `0.0.0.0` in `server/index.ts`. Nothing works.

```
const port = process.env.PORT ? Number(process.env.PORT) : 3000
const host = '0.0.0.0'
app.listen(port, host)
console.log(`Server running at http://${host}:${port}`)
```

Also tried using the same host in `vite.config.ts`.

```
import react from '@vitejs/plugin-react'
import vike from 'vike/plugin'
import { UserConfig } from 'vite'

const config: UserConfig = {
    server: {
        host: '0.0.0.0',
        hmr: {
            host: '0.0.0.0'
        }
    },
    plugins: [react(), vike()]
}

export default config
```

Running on Ubuntu 22.04 LTS using Windows WSL Feature.

Please help me.

**Comment by isakura313** [comment]
Any other node project working correctly?

**Comment by axelthat** [comment]
Yes. Here is a simple express project:

![image](https://github.com/vikejs/vike/assets/31353695/b32c91f2-7ece-4da4-9a2a-ba664ab4e417)

![image](https://github.com/vikejs/vike/assets/31353695/48f03e9d-e512-4fe4-b399-e1ab408f0d2f)

**Comment by isakura313** [comment]
Good. How do you create you'r barebones project? It's good if you provide steps or repo itself.

**Comment by axelthat** [comment]
I just followed instructions from here: https://vike.dev/new `npm create vike@latest`
I added nothing, removed nothing.

**Comment by isakura313** [comment]
Did you use Bati for scaffold?  And can you please build  and run by: npm run build && node server . We need to know it's dev mode or more

**Comment by axelthat** [comment]
This is all what I did.
https://drive.google.com/file/d/1oNwbJx5tWnNq1AJGTmS9wD39WvyiPgp3/view?usp=sharing

**Comment by isakura313** [comment]
ok, got it. Well, try Bati with h3, i think it can help you. Now i don't have wsl for this tests

**Comment by axelthat** [comment]
@isakura313 I am sorry I don't follow. What is "Bati with h3"?

**Comment by isakura313** [comment]
@axelthat  https://batijs.dev/ 
<img width="1398" alt="image" src="https://github.com/vikejs/vike/assets/18578858/e6288c1b-92c2-4e86-9f44-883422101b09">

**Comment by brillout** [comment]
This doesn't seem related to Vike, but feel free to let me know if you have evidence it's Vike related.
